src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SuspiciousCastFromCharToInt.Fixer.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+using RequiredSymbols = Microsoft.NetCore.Analyzers.Runtime.SuspiciousCastFromCharToIntAnalyzer.RequiredSymbols;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class SuspiciousCastFromCharToIntFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(SuspiciousCastFromCharToIntAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var doc = context.Document;
+            var token = context.CancellationToken;
+            var root = await doc.GetSyntaxRootAsync(token).ConfigureAwait(false);
+            var model = await doc.GetSemanticModelAsync(token).ConfigureAwait(false);
+
+            if (model.GetOperation(root.FindNode(context.Span), token) is not IArgumentOperation argumentOperation)
+                return;
```

Comment:
Could this be parenthesized operation (for example)?

Reply:
@Youssef1313 No. `FindNode` always finds the outermost node, unless `getInnermostNodeForTie` is specified as `true`. If a parenthesized operation were passed as the argument, it would be the `Value` property of the argument operation. I will check locations that access the `Value` property; you're right that I may need some calls to `WalkDownParentheses` in those locations.